library/old_protocol/serialize.cpp

```diff
@@ -0,0 +1,97 @@
+#include "serialize.hpp"
+
+#include <msgpack.hpp>
+
+#include "library/elliptics.h"
+
+namespace msgpack {
+
+using namespace ioremap::elliptics;
+
+template <typename Stream>
+inline msgpack::packer<Stream> &operator <<(msgpack::packer<Stream> &o, const dnet_time &v) {
+	o.pack_array(2);
+	o.pack_fix_uint64(v.tsec);
+	o.pack_fix_uint64(v.tnsec);
+	return o;
+}
+
+template <typename Stream>
+inline msgpack::packer<Stream> &operator<<(msgpack::packer<Stream> &o, const n2::lookup_response &v) {
+	o.pack_array(12);
+
+	o.pack(v.record_flags);
+	o.pack(v.user_flags);
+	o.pack(v.path);
+
+	o.pack(v.json_timestamp);
+	o.pack(v.json_offset);
+	o.pack(v.json_size);
+	o.pack(v.json_capacity);
+
+	o.pack(v.data_timestamp);
+	o.pack(v.data_offset);
+	o.pack(v.data_size);
+
+	o.pack(v.json_checksum);
+	o.pack(v.data_checksum);
+
+	return o;
+}
+
+} // namespace msgpack
+
+namespace ioremap { namespace elliptics { namespace n2 {
+
+void enqueue_net(dnet_net_state *st, net_iovec serialized) {
+	std::unique_ptr<n2_net_iovec> iov(new n2_net_iovec({ std::move(serialized) }));
```

Comment:
Why you use two pairs of brackets?
And in `n2_make_repliers_via_request_queue`.

Reply:
Fixed.